### PR TITLE
Updated NYC/MA required and optional fields

### PR DESCRIPTION
**Pull request by GeorgeCodes19**
Closes [FFS-1300](https://jiraent.cms.gov/browse/FFS-1300)

## Changes

- Updated `snap_application_date` to be required for all agencies
- `client_id_number` is now optional for NYC
- `middle_name` is now optional for all agencies

## Testing

I've tested things locally and created tests to assert that the requirements are met

### CODE REVIEW

File: app/db/migrate/20240814192234_change_requirements_for_cbv_flow_invitations_fields.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ChangeRequirementsForCbvFlowInvitationsFields < ActiveRecord::Migration[7.1]
+  def change
+    change_column_null :cbv_flow_invitations, :middle_name, true
+    change_column_null :cbv_flow_invitations, :snap_application_date, true
```
Comment (GeorgeCodes19) [inline app/db/migrate/20240814192234_change_requirements_for_cbv_flow_invitations_fields.rb:4]: Need to fix this

File: app/spec/controllers/cbv_flow_invitations/ma_spec.rb [hunk 2]
```diff
@@ -29,15 +29,12 @@
...
-      agency_id_number: "789012"
-    } }
-
     let(:cbv_flow_invitation_params) do
-      attributes_for(:cbv_flow_invitation, custom_params)
+      attributes_for(:cbv_flow_invitation,
+             site_id: "ma",
+             beacon_id: "ABC123",
+             agency_id_number: "789012",
+             )
```
Comment (tdooner) [inline app/spec/controllers/cbv_flow_invitations/ma_spec.rb:37]: 7 space indent?? you really gotta figure out why your editor is butchering this indentation, lol

File: app/spec/controllers/cbv_flow_invitations/ma_spec.rb [hunk 3]
```diff
@@ -56,5 +53,13 @@
...
       expect(invitation.beacon_id).to eq("ABC123")
     end
+
+    it "requires the ma fields" do
+      post :create, params: {
+        secret: invite_secret,
+        site_id: ma_params[:site_id],
+        cbv_flow_invitation: cbv_flow_invitation_params
+      }
+    end
```
Comment (tdooner) [inline app/spec/controllers/cbv_flow_invitations/ma_spec.rb:63]: This test is missing an `expect`. What should it be testing? That the invitation was created? Isn't that the same as the previous test? Should this be testing the optional params like the NYC test below?

**Review by tdooner — APPROVED** [review]
Looks good. I cross-referenced the logic with the table in the ticket, and it looks good.

File: app/spec/controllers/cbv_flow_invitations/ma_spec.rb (hunk 3 above)
Comment (GeorgeCodes19) [reply app/spec/controllers/cbv_flow_invitations/ma_spec.rb:63]: definitely missing the assertion!

File: app/spec/controllers/cbv_flow_invitations/ma_spec.rb (hunk 2 above)
Comment (GeorgeCodes19) [reply app/spec/controllers/cbv_flow_invitations/ma_spec.rb:37]: It's been killing me. I'll see how to properly configure the Rubymine linter after this PR.

**Pull request merged**